polaris-service/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1826,6 +1826,27 @@ private void updateTableLike(TableIdentifier identifier, PolarisEntity entity) {
 
     List<PolarisEntity> catalogPath = resolvedEntities.getRawParentPath();
     PolarisEntity leafEntity = resolvedEntities.getRawLeafEntity();
+
+    // Check that purge is enabled, if it is set:
+    if (catalogPath != null && !catalogPath.isEmpty() && purge) {
+      boolean dropWithPurgeEnabled =
+          callContext.getPolarisCallContext()
+              .getConfigurationStore()
+              .getConfiguration(
+                  callContext.getPolarisCallContext(),
+                  (CatalogEntity) catalogPath.getFirst(),
```

Comment:
For consistency we can use the `catalogEntity` private member that BasePolarisCatalog already has since other methods would also be using that for config overrides in situations where a catalogPath might not be resolved/present for whatever reason.